nitrite/src/main/java/org/dizitart/no2/Nitrite.java

```diff
@@ -145,6 +145,34 @@ public NitriteCollection getCollection(String name) {
         return null;
     }
 
+    /**
+     * Opens a type-safe object repository with a key identifier from the store. If the repository
+     * does not exist it will be created automatically and returned. If a
+     * repository is already opened, it is returned as is.
+     * 
+     * [icon="{@docRoot}/note.png"]
+     * NOTE: Returned repository is thread-safe for concurrent use.
+     *
+     * @param <T>  the type parameter
+     * @param key  the key that will be appended to the repositories name
+     * @param type the type of the object
+     * @return the repository containing objects of type {@link T}.
+     * @see ObjectRepository
+     */
+    public <T> ObjectRepository<T> getRepository(String key, Class<T> type) {
+        if (store != null) {
+            String name = findObjectStoreName(key, type);
+            NitriteMap<NitriteId, Document> mapStore = store.openMap(name);
+            NitriteCollection collection = CollectionFactory.open(mapStore, context);
+            ObjectRepository<T> repository = RepositoryFactory.open(type, collection, context);
+            context.getRepositoryRegistry().add(type);
```

Comment:
You are adding the repository type to the repository registry also you can never get this repository with `getRepository(type)`. I think this will cause problems in `Nitrite.closeCollections()`, because it iterates over all types and tries to close the repositories. For this type it will not find a repository. It is null safe, so it should not cause an error, but the repositories using a key will not be closed by `Nitrite.closeCollections()` and you have to be careful to always close them yourself before closing the database. I think that is a bit dangerous.

Reply:
@reboot I have not merged @LarsWerkman 's changes as is. I had to modify the repositoryRegistry type from set to map to handle these scenarios. Please have a look into the latest `Nitrite` class on master.

Thanks for reviewing the code though. Really appreciate it.

Reply:
Oh, i should have checked. Thanks for letting me know.